### ISSUE TITLE
[Bug] WithAuthority ignores WithTenantID

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-authentication-library-for-dotnet#1320]
**Which Version of MSAL are you using ?**
4.2.1

**Platform**
all

**What authentication flow has the issue?**
public client builder
possibly the acquireToken* API as well

**Repro**

```csharp
IPublicClientApplication pca = PublicClientApplicationBuilder
                            .Create(<client-id>)
                            .WithAuthority(AadAuthorityAudience.AzureAdMyOrg)
                            .WithTenantId("49f548d0-12b7-4169-a390-bb5304d24462")
                         
                            .Build();
```

**Expected behavior**
PCA should be constructed - it has enough info.

**Actual behavior**
InvalidOperationException (!!) with message: 

`System.InvalidOperationException: 'When specifying AadAuthorityAudience.AzureAdMyOrg, you must also specify a tenant domain or tenant guid.'`

**Comment by motionsmith** [comment]
Is there a workaround?

**Comment by bgavrilMS** [comment]
@motionsmith - yes, you can use 

` .WithAuthority(AzureCloudInstance.AzurePublic, Guid.Parse("1234.."))` or 
`.WithAuthority("https://login.microsoftonline.com/1234...")`

**Comment by jennyf19** [comment]
Included in [4.6 release](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.6)
